### PR TITLE
[Popover] Fix height measuring issue

### PR DESCRIPTION
**Pull request by martenbjork**
Ensures that the size of Popovers update when the size of child elements changes.

Before, it would only be detected if the children's DOM nodes changed. With this change, updates to the `style` attribute is also detected an triggers a resize of the Popover.

## Performance

I've stress tested the new code by rendering 50 popovers that constantly redraws every 1000 / 60 ms. It renders in 60fps without dropping any frames.

https://user-images.githubusercontent.com/875708/233351722-7061aadd-48cc-46fe-994e-1db141e99c0b.mov

## Avoiding unnecessary rerenders

When a child's `style` attribute changes, the `Popover` rerenders as well to allow it to resize itself:

https://user-images.githubusercontent.com/875708/233351988-02dd60f4-89e4-412a-9c58-eb702351e10b.mov

However, when another attribute changes, the `Popover` ignores it in order to avoid expensive rerenders:

https://user-images.githubusercontent.com/875708/233352103-4e91a4f1-5061-4a78-bd70-ee91a8099b60.mov

### CODE REVIEW

File: polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx [hunk 1]
```diff
@@ -63,6 +63,7 @@ const OBSERVER_CONFIG = {
   childList: true,
   subtree: true,
   characterData: true,
+  attributes: true,
```
Comment (martenbjork) [inline polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx]: Track changes to attributes, like `style={{height}}`

Comment (kyledurand) [reply polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx]: I think this is ok but don't have much context. @AndrewMusgrave can you think of any issues we might run into with this or maybe performance concerns?

Comment (jesstelford) [inline polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx]:
```suggestion
  // Track changes to attributes, like `style={{height}}` which is useful when multi-line text areas are resized, etc.
  attributes: true,
```

**Review by jesstelford — APPROVED** [review]
LGTM

**Comment by jesstelford** [comment]
Needs a changeset - minor, I think.

File: polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx (hunk 1 above)
Comment (martenbjork) [reply polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx]: Added some additional performance info in the issue description.

Comment (AndrewMusgrave) [reply polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx]: Not that I'm aware of 😄

**Pull request merged**